File: MainSolution/Hospital/Shared Model/Model/Room.cs
namespace Model
{
    public class Room
    {
        public RoomType RoomType { get; set; }
        public int Id { get; set; }
        public string Name { get; set; }
        public bool Available { get; set; }
        public Room(RoomType rt, int id, string name, bool a)
        {
            this.RoomType = rt;
            this.Id = id;
            this.Name = name;
            this.Available = a;
        }

        public Room()
        {

        }
        public override string ToString()
        {
            return Id + " | " + Name;
        }

        public bool IsAppointmentRoom()
        {
            if (RoomType == RoomType.APPOINTMENT_ROOM)
                return true;

            return false;
        }
    }
}

File: WebClients/ManagerWebClient/src/app/interfaces/room.ts
import { RoomPosition } from "../model/room-position.model";

export enum RoomType {
  AppointmentRoom,
  OperationRoom,
  Storage,
  Bedroom,
  OfficeRoom,
}

export class Room {
    id! : number;
    name! : string;
    description! : string;
    width! : number;
    height! : number;
    floorNumber! : number;
    buildingName! : string;
    doctors! : [];
    roomInventories! : [];
    scheduledEvents! : [];
    roomType: number;
    roomPosition: RoomPosition;
}
